### PR TITLE
Add Ubuntu22.04 build, simplify workflow, and cleanup a couple things in the script 

### PR DESCRIPTION
**Pull request by L1ghtmann**
Tested successfully on both [osx ](https://github.com/L1ghtmann/swift-toolchain-linux/actions/runs/3504920600) and [linux](https://github.com/L1ghtmann/swift-toolchain-linux/actions/runs/3506361523)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,53 +5,51 @@ on:
...
-      with:
-        name: upload_url
-        path: upload_url.txt
-
   build:
-    needs: create_release
     strategy:
       matrix:
-        os: [ubuntu20.04-aarch64, ubuntu20.04, ubuntu18.04]
+        os: [ubuntu22.04, ubuntu20.04, ubuntu18.04]
```
Comment (kabiroberai) [inline .github/workflows/release.yml]: We should have aarch64 builds, ideally add them back for both 20.04 and 22.04

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -5,53 +5,51 @@ on:
...
+        draft: true
+
+    - name: Attach Builds To Release
+      run: |
+        set -x
+        toolchains=()
+        for tc in **/*.tar.xz; do
+          toolchains+=("-a" "$tc")
+        done
+        hub release edit "${toolchains[@]}" -m "Draft Release" "$TAG"
```
Comment (kabiroberai) [inline .github/workflows/release.yml]: It's preferable to have each toolchain be added to the release as it builds, so that one build failing doesn't affect all of them. Using `hub` and `$GITHUB_ENV` are good changes, but if you could maintain the incremental behaviour that would be great.

File: create-toolchain [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+#!/usr/bin/env bash
 set -e
 cd "$(dirname "$0")"
 
 DARWIN_TOOLS_VERSION="2.2.0"
 
-[[ "$(uname -s)" = "Darwin" ]] && is_mac=true
+if [[ "$(uname -s)" = "Darwin" ]]; then
+	is_mac=1
+fi
```
Comment (kabiroberai) [inline create-toolchain]: Is there any reason for this change besides personal stylistic preference? If that's the only reason then consider reverting this change to minimize the diff.

File: create-toolchain [hunk 4]
```diff
@@ -98,11 +94,15 @@ function link_file {
...
 
+info "Modifying Linux toolchain"
 # symlink all files in these dirs from the host toolchain...
 find linux/host \( -path 'linux/host/bin/*' -o -path 'linux/host/local/*' -o -path 'linux/host/share/*' -o -path 'linux/host/lib/*' \) -and -not -type d -print0 | xargs -0 -I{} bash -c 'link_file "$@"' _ {}
 # ...but replace some components with mac ones...
 rm -rf linux/iphone/lib/swift{,_static}
-cp -a "${mac_toolchain}"/usr/lib/{swift,swift-5.0,swift-5.5,arc} linux/iphone/lib/
+if [[ "${is_mac}" -eq 1 ]]; then
+	cp -a "${mac_toolchain}"/usr/lib/{swift-5.0,arc} linux/iphone/lib/
+fi
```
Comment (kabiroberai) [inline create-toolchain]: What's the rationale behind this change? arclite is arguably less relevant now, but including swift-5.0 is still necessary if back-deploying to iOS < 12.2.

**Review by kabiroberai — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left a few comments, some stylistic nits but others are substantial, so it'd be great if you could address those.

File: .github/workflows/release.yml (hunk 1 above)
Comment (L1ghtmann) [reply .github/workflows/release.yml]: I may be missing something, but I don't believe there are runners for anything other than x64. To support aarch64, an additional action will have to be checked out ([example](https://github.com/uraimo/run-on-arch-action)). I can look into that if aarch64 is still desired.

File: create-toolchain (hunk 3 above)
Comment (L1ghtmann) [reply create-toolchain]: Are you referring to the true/false -> 1/0 or the one-liner being written out? I wrote out the one-liners for clarity and believe I changed the true/false for my own sake while testing. I can revert both if you'd like.

File: create-toolchain (hunk 4 above)
Comment (L1ghtmann) [reply create-toolchain]: They're not included in the OSX toolchain provided by Apple -- https://github.com/L1ghtmann/swift-toolchain-linux/actions/runs/3506320032/jobs/5873283058#step:3:5360

File: .github/workflows/release.yml (hunk 1 above)
Comment (kabiroberai) [reply .github/workflows/release.yml]: This isn't the runner OS, it's the target OS — the create-toolchain action should still run on macos-12.

Comment (L1ghtmann) [reply .github/workflows/release.yml]: Right, it does, but the Darwin tools are compiled on each respective OS. So, without a way to compile those for aarch64, adding an aarch64 target here will result in the curl of a nonexistent Darwin tools tarball. 

I apologize if I'm missing something here. I was admittedly a bit confused how there was an aarch64 Darwin tools build in your latest release when there doesn't seem to be a mention of it anywhere in the repo's workflow or in the build script.

Comment (kabiroberai) [reply .github/workflows/release.yml]: mb for not clarifying, I built the aarch64 toolchain on my M1 machine — the Dockerfile in the darwin-tools repo is multi-arch. This repo doesn't face the same issue though, so aarch64 in the GH action should be okay (at least when darwin-tools-linux has a corresponding toolchain).

File: create-toolchain (hunk 3 above)
Comment (kabiroberai) [reply create-toolchain]: both, ty

File: create-toolchain (hunk 4 above)
Comment (kabiroberai) [reply create-toolchain]: hmm interesting, thanks for flagging — in that case we should either emit a warning that the toolchain won't include pre-12.2 compat, or not support building on Linux. There's also a third option which is getting the 5.0 libs from somewhere else (since they're effectively frozen in time now) but I'm not sure what a good official source would be.

Comment (kabiroberai) [reply create-toolchain]: re the third option, we might be able to extract them from https://download.swift.org/swift-5.0-release/xcode/swift-5.0-RELEASE/swift-5.0-RELEASE-osx.pkg

File: .github/workflows/release.yml (hunk 1 above)
Comment (kabiroberai) [reply .github/workflows/release.yml]: Added ubuntu22.04 (x64 and aarch64) packages to the current release, so ubuntu22.04 and ubuntu22.04-aarch64 should work now

Comment (L1ghtmann) [reply .github/workflows/release.yml]: Re-added the aarch64 targets.

File: create-toolchain (hunk 3 above)
Comment (L1ghtmann) [reply create-toolchain]: Reverted bool and one-liner changes.

File: .github/workflows/release.yml (hunk 2 above)
Comment (L1ghtmann) [reply .github/workflows/release.yml]: Implemented incremental behavior but switched to `gh` cli as `hub` appeared to have a couple of issues.

File: create-toolchain (hunk 4 above)
Comment (L1ghtmann) [reply create-toolchain]: Unsure which 5.0 directories need to be included, so just went with a warning for the time being.

**Review by kabiroberai — APPROVED** [review]
Love the release.yml cleanup 🙌
